cpp/src/umap/simpl_set_embed/algo.cuh

```diff
@@ -168,21 +168,30 @@ T create_rounding_factor(T max_abs, int n)
 }
 
 template <typename T, typename nnz_t>
-T create_gradient_rounding_factor(
-  const int* head, nnz_t nnz, int n_samples, T alpha, rmm::cuda_stream_view stream)
+uint32_t get_95p_n_edges(const int* head, nnz_t nnz, int n_samples, rmm::cuda_stream_view stream)
 {
   rmm::device_uvector<T> buffer(n_samples, stream);
-  // calculate the maximum number of edges connected to 1 vertex.
+
+  // Count edges per vertex
   thrust::reduce_by_key(rmm::exec_policy(stream),
                         head,
                         head + nnz,
                         thrust::make_constant_iterator(1u),
                         thrust::make_discard_iterator(),
                         buffer.data());
-  auto ptr         = thrust::device_pointer_cast(buffer.data());
-  uint32_t n_edges = *(thrust::max_element(rmm::exec_policy(stream), ptr, ptr + buffer.size()));
-  T max_abs        = T(n_edges) * T(4.0) * std::abs(alpha);
-  return create_rounding_factor(max_abs, n_edges);
+
+  // Sort the buffer
+  thrust::sort(rmm::exec_policy(stream), buffer.data(), buffer.data() + n_samples);
```

Comment:
@viclafargue have you done any benchmarking on this to evaluate the impact? I understand this is fixing a bug that surfaces only sometimes- but if there's a major perf cost here then we'd be paying it even if in the case it's not present, right? 

We should try and better characterize this so that we can 1) test it, and 2) determine when it's needed.